projects/plugins/jetpack/extensions/blocks/goodreads/edit.js

```diff
@@ -0,0 +1,419 @@
+					/>
+
+					<TextControl
+						label={ __( 'Number of books', 'jetpack' ) }
+						type="number"
+						inputMode="numeric"
+						min="1"
+						max={ style === 'grid' ? 200 : 100 }
+						value={ bookNumber || 5 }
+						onChange={ value => setAttributes( { bookNumber: value } ) }
+					/>
+
+					{ style === 'default' && this.renderDisplaySettings() }
+				</PanelBody>
+			</InspectorControls>
+		);
+	}
+
+	renderEditEmbed() {
+		const { className, noticeUI } = this.props;
+		const { userInput } = this.state;
+
+		return (
+			<div className={ className }>
+				<Placeholder
+					label={ __( 'Goodreads', 'jetpack' ) }
+					instructions={ __( 'Paste a link to a Goodreads profile.', 'jetpack' ) }
+					icon={ <BlockIcon icon={ icon } /> }
+					notices={ noticeUI }
+				>
+					<form onSubmit={ this.submitForm }>
+						<input
+							type="url"
+							value={ userInput }
+							className="components-placeholder__input"
+							aria-label={ __( 'Goodreads profile URL', 'jetpack' ) }
+							placeholder={ __( 'Enter a Goodreads profile URL to embed here…', 'jetpack' ) }
+							onChange={ event => this.setState( { userInput: event.target.value } ) }
+						/>
+						<Button variant="secondary" type="submit">
+							{ _x( 'Embed', 'submit button label', 'jetpack' ) }
+						</Button>
+					</form>
+				</Placeholder>
+			</div>
+		);
+	}
+
+	renderBlockControls() {
+		return (
+			<BlockControls>
+				<ToolbarButton
+					className="components-toolbar__control"
+					label={ __( 'Edit URL', 'jetpack' ) }
+					icon="edit"
+					onClick={ () => this.setState( { displayPreview: false } ) }
+				/>
+			</BlockControls>
+		);
+	}
+
+	renderInlinePreview() {
+		const { className } = this.props;
+		const { goodreadsId, link, id } = this.props.attributes;
+
+		if ( ! goodreadsId ) {
+			return;
+		}
+
+		const html = `
+		<style> [class^=gr_custom_container_] { border: 1px solid gray; border-radius: 10px; margin: auto; padding: 0 5px 10px 5px; background-color: #FFF; color: #000; width: 300px; }  [class^=gr_custom_header_] { border-bottom: 1px solid gray; width: 100%; padding: 10px 0; margin: auto; text-align: center; font-size: 120%; }  [class^=gr_custom_each_container_] { width: 100%; clear: both; margin: auto; overflow: auto; padding-bottom: 4px; border-bottom: 1px solid #aaa; }  [class^=gr_custom_each_container_] { width: 100%; clear: both; margin-bottom: 10px; overflow: auto; padding-bottom: 4px; border-bottom: 1px solid #aaa; }  [class^=gr_custom_book_container_] { overflow: hidden; height: 60px; float: left; margin-right: 6px; width: 39px; }  [class^=gr_custom_author_] { font-size: 10px; }  [class^=gr_custom_tags_] { font-size: 10px; color: gray; }  [class^=gr_custom_rating_] { float: right; }  [class^=gr_grid_book_container] { float: left; width: 98px; height: 160px; padding: 0px 0px; overflow: hidden; }  a { text-decoration: none; }  a:hover { text-decoration: underline; } img { max-width: 100%; }</style>
```

Comment:
There has got to be a better way of doing this...but no idea what that would be!